sources/osquery_thrift.js

```diff
@@ -30,6 +38,52 @@ module.exports = class OSQuery {
     }
   }
 
+  static start() {
+    const socket = socketPath[platform] + path.sep + 'shell.em'
+    const binary = osqueryPlatforms[platform]
+    const prefix = process.env.NODE_ENV === 'development' ? '' : '../'
+    const osqueryPath = path.resolve(__dirname, prefix + binary)
+
+    log.info(`startup: ${osqueryPath, ['--nodisable_extensions', `--extensions_socket=${socket}`].join(' ')}`)
+
+    return new Promise((resolve, reject) => {
+      const osqueryi = spawn(osqueryPath, ['--nodisable_extensions', `--extensions_socket=${socket}`], {
+        windowsHide: true,
+      })
+
+      setTimeout(() => {
+        this.connection = ThriftClient.getInstance({ path: socket })
+        resolve()
+      }, 100)
+
+      osqueryi.stderr.on('data', (data) => {
+        log.error(`osquery process stderr: ${data}`);
+      });
+
+      osqueryi.on('error', (err) => {
+        if (this.connection) {
+          this.connection.end()
+        }
+        log.error(`osquery execution error: ${err}`)
+        reject(new Error(`Unable to spawn osquery: ${err}`))
```

Comment:
Is this the only error case? Is it possible to crash osqueryi with a bad query, in which case we should try to bring it back up?

Reply:
From my tests so far, this is the only error case. The bad queries I've tested catch and report syntax errors and don't kill the osqueryi process.

```bash
osquery> select $%^&*( from bobby_tables;
Error: unrecognized token: "$"
osquery> i can haz segfault?;
Error: near "i": syntax error
osquery>
```

I think it does make sense to cover ourselves should osquery crash for an unexpected reason